### PR TITLE
Cw 598 fixes for electrum based wallets

### PR DESCRIPTION
**Pull request by rafael-xmr**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [X] Initial Manual Tests Passed
- [X] Double check modified code and verify it with the feature/task requirements
- [X] Format code
- [X] Look for code duplication
- [X] Clear naming for variables and methods

### CODE REVIEW

File: cw_bitcoin/lib/electrum.dart [hunk 1]
```diff
@@ -243,22 +245,13 @@ class ElectrumClient {
...
-              headers: <String, String>{'Content-Type': 'application/json; charset=utf-8'},
-              body: transactionRaw)
-          .then((http.Response response) {
-        if (response.statusCode == 200) {
-          return response.body;
-        }
-
-        throw Exception('Failed to broadcast transaction: ${response.body}');
-      });
-    }
```
Comment (OmarHatem28) [inline cw_bitcoin/lib/electrum.dart:259]: why remove testnet api?

File: cw_bitcoin/lib/electrum_wallet.dart [hunk 2]
```diff
@@ -898,6 +918,10 @@ class EstimatedTxResult {
 }
 
 BitcoinBaseAddress addressTypeFromStr(String address, BasedUtxoNetwork network) {
+  try {
+    return BitcoinCashAddress(address).baseAddress;
```
Comment (OmarHatem28) [inline cw_bitcoin/lib/electrum_wallet.dart:922]: let's put the possibility of the address being able to be parsed to bitcoin cash address while it's not a bitcoin cash address, so let's be on the safe side and also add a check if the network is bitcoin cash

File: lib/bitcoin/cw_bitcoin.dart [hunk 3]
```diff
@@ -122,12 +122,29 @@ class CWBitcoin extends Bitcoin {
...
+      final sk = ECPrivate.random();
+      final electrumWallet = wallet as ElectrumWallet;
+
+      if (wallet.type == WalletType.bitcoinCash) {
+        final p2pkhAddr = sk.getPublic().toP2pkhAddress();
+        final estimatedTx = await electrumWallet.estimateTxFeeAndInputsToUse(
+            0,
+            true,
+            // Deposit address + change address
+            [p2pkhAddr, p2pkhAddr],
```
Comment (OmarHatem28) [inline lib/bitcoin/cw_bitcoin.dart:135]: why do we have a change address when it should send all?

File: cw_bitcoin/lib/pending_bitcoin_transaction.dart [hunk 4]
```diff
@@ -35,10 +35,19 @@ class PendingBitcoinTransaction with PendingTransaction {
...
+
+    final result = await electrumClient.broadcastTransaction(
+        transactionRaw: hex, network: network, idCallback: (id) => callId = id);
 
     if (result.isEmpty) {
-      throw BitcoinCommitTransactionException();
+      if (callId != null) {
+        if (electrumClient.getErrorMessage(callId!).contains("dust")) {
+          errorMessage = "Transaction rejected. Output cannot be dust. Please increase the amount.";
+        }
```
Comment (OmarHatem28) [inline cw_bitcoin/lib/pending_bitcoin_transaction.dart:48]: while testing, let's monitor other error messages that might happen and add a meaningful message for them too

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: cw_bitcoin/lib/electrum.dart (hunk 1 above)
Comment (rafael-xmr) [reply cw_bitcoin/lib/electrum.dart]: Not needed anymore to send txs, and it was used to log the response but new error messages by id can be used instead

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**